src/rules/indentation.js

```diff
@@ -16,60 +16,86 @@ var availableConfigs = {
   'but': 2
 };
 
-var errors = [];
+var availableConfigs = _.merge({}, defaultConfig, {
```

Comment:
With this change, we have 2 configuration fallback mechanisms (one for step types and one for tag types) that are coded in different ways and different places. I think we should make the fallbacks work more similarly. To that I suggest we:
- move the entries for `given`, `when`, `then`, `and`, `but` from `defaultConfig` to `availableConfigs`
- move the config fallback logic for `scenario tag` and `feature tag` from `mergeConfiguration` right before the testTags calls eg:
```
var featureTagConfig = 'feature tag' in mergedConfiguration ? 'feature tag' : 'Feature';
testTags(feature.tags, featureTagConfig)
```
This is similar to what we do for the step type config.

Reply:
If I understand correctly what you're asking for, this would change the error messages?  For example, with the code as you describe you would get an error message like:

`Wrong indentation for "Scenario", expected indentation level of 0, but got 1`

Whereas with my code you would get the error message:

`Wrong indentation for "scenario tag", expected indentation level of 0, but got 1`

Which could be confusing to the user.

This was the reason for the two types of defaulting - one affects the error message but the other does not.  I think at the time I did think about changing the fallback logic for the step defs instead (currently near L53), because I agree it's a bit weird having two types of defaulting, but that would change those error messages instead, which I didn't want to do either.

Reply:
I completely forgot about the error messages :c I agree that it would be confusing so it's not worth trying to merge the 2 configurations